site/app/controllers/SocketController.php

```diff
@@ -0,0 +1,20 @@
+<?php
+    namespace app\controllers;
+
+    use Ratchet\Server\IoServer;
+    use Ratchet\Http\HttpServer;
+    use Ratchet\WebSocket\WsServer;
+    use app\libraries\Socket;
+
+    require dirname(__DIR__) . '/../vendor/autoload.php';
```

Comment:
require_once may be better. The require_once statement is identical to require except PHP will check if the file has already been included, and if so, not include (require) it again.

https://www.php.net/manual/en/function.require-once.php

Reply:
I'm not sure if that is what we need here, I guess I just copied that line from another place within Submitty :o

Reply:
Where is it? `index.php` is using `require_once` though...